process/block/preprocess/transactions.go

```diff
@@ -422,8 +423,24 @@ func (txs *transactions) processTxsToMe(
 		if err != nil {
 			return err
 		}
+
+		gasConsumedByTx, moveBalanceGas := txs.gasHandler.GasConsumed(txHash), txs.economicsFee.ComputeGasLimit(tx)
```

Comment:
This mechanism from L427-L442 should be added also in process txs dest me in the proposer side